Testing/2.0.8

Croydon opened the pull request.

Croydon committed:
Apply latest conventions and CI templates

Croydon committed:
Correct license attribute

uilianries reviewed (approved):
LGTM

Croydon commented:
Urgh. @uilianries Any ideas where this is coming from? https://travis-ci.org/bincrafters/conan-sdl2/jobs/449155334#L1758

uilianries commented:
looks like is related to python2.7-minimal package. conanio Docker images are running /usr/bin/python as python 3.7. Maybe will be necessary install python2.7 full.

Croydon commented:
It only fails in the GCC 4.9 containers (with Python 3.6) 🤔

uilianries commented:
Both gcc-4.8 and gcc-4.9 use old Ubuntu distros, I used some workarounds to install python3.6. going to check now. thanks for reporting

elizagamedev committed:
Fix sdl2-config behavior

There is a bug in SDL2's CMakeLists.txt which tries to copy libSDL2.so
from an incorrect location. This behavior had previously been patched
out, which had the unfortunate side effect of breaking the behavior of
sdl2-config. The reason it was patched out appears to be because the
CMakeLists.txt does not handle debug-only builds correctly, as it
appends a "d" postfix to the libraries, which trips up both the symlink
and sdl2-config. This behavior has been corrected by both patching out
the debug postfix and correcting the path of the symlink.

elizagamedev committed:
Fix macOS build

elizagamedev committed:
Update conventions

SSE4 commented on conanfile.py:
```diff
@@ -197,7 +203,14 @@ def source(self):
         tools.get(source_url)
         extracted_dir = "SDL2-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+<<<<<<< HEAD
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -197,7 +203,14 @@ def source(self):
+                os.path.join(self._source_subfolder, 'CMakeLists.txt'),
+                'install(FILES ${SDL2_BINARY_DIR}/libSDL2.${SOEXT} DESTINATION "lib${LIB_SUFFIX}")',
+                '')
+=======
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -197,7 +203,14 @@ def source(self):
+                '')
+=======
         tools.patch(base_path=self._source_subfolder, patch_file="cmake.patch")
+>>>>>>> stable/2.0.8
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -265,16 +278,25 @@ def configure_cmake(self):
         elif self.settings.os == "Windows":
             cmake.definitions["DIRECTX"] = self.options.directx
 
+<<<<<<< HEAD
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -265,16 +278,25 @@ def configure_cmake(self):
 
+<<<<<<< HEAD
+        cmake.configure(build_dir=self._build_subfolder)
+=======
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -265,16 +278,25 @@ def configure_cmake(self):
         cmake.build()
 
     def package(self):
+<<<<<<< HEAD
```
conflict marker

SSE4 commented on conanfile.py:
```diff
@@ -265,16 +278,25 @@ def configure_cmake(self):
+=======
         cmake = self.configure_cmake()
         cmake.install(build_dir=os.path.join(self.build_folder, self._build_subfolder))
+>>>>>>> stable/2.0.8
```
conflict marker

SSE4 reviewed (changes requested):
this one seems to be messed up, there are conflict markers within files, please fix it

SSE4 commented:
@Croydon there are conflict markers within conanfile, please fix it

Croydon commented:
@SSE4 Please read the pull request history. I didn't even push this and don't know what this is about

SSE4 commented:
@Croydon so what should we do? revert last commit?

Croydon commented:
@solvingj ^

solvingj commented:
Looks like was a mistake. Apologies, need to look more closely.

Commit:
Fix sdl2-config libraries on macOS

Previously, CMakeLists.txt was generating an sdl2-config script on macOS
that attempted to link frameworks with arguments such as
`-l/Library/.../CoreAudio.framework`, which is obviously erroneous. The
arguments have been corrected to the format `-framework CoreAudio`, etc.
